docker/context.go

```diff
@@ -0,0 +1,444 @@
+	}
+
+	var contexts []ContextInfo
+	// Parse JSON lines (each line is a separate JSON object)
+	lines := []byte{}
+	for _, b := range output {
+		if b == '\n' {
+			if len(lines) > 0 {
+				var ctx struct {
+					Name           string `json:"Name"`
+					Current        bool   `json:"Current"`
+					Description    string `json:"Description"`
+					DockerEndpoint string `json:"DockerEndpoint"`
+				}
+				if err := json.Unmarshal(lines, &ctx); err != nil {
+					return nil, fmt.Errorf("failed to parse context JSON: %w", err)
+				}
+
+				// Check if TLS is enabled by inspecting the context
+				usesTLS := false
+				if inspectJSON, err := InspectContext(ctx.Name); err == nil {
+					// Parse as array first since docker context inspect returns an array
+					var inspectArray []map[string]interface{}
+					if err := json.Unmarshal([]byte(inspectJSON), &inspectArray); err == nil && len(inspectArray) > 0 {
+						inspectData := inspectArray[0]
+						// Check for TLSMaterial field (top level)
+						if tlsMaterial, ok := inspectData["TLSMaterial"].(map[string]interface{}); ok {
+							if dockerTLS, ok := tlsMaterial["docker"]; ok && dockerTLS != nil {
+								usesTLS = true
+							}
+						}
+						// Also check legacy TLSData field in Endpoints
+						if !usesTLS {
+							if endpoints, ok := inspectData["Endpoints"].(map[string]interface{}); ok {
+								if docker, ok := endpoints["docker"].(map[string]interface{}); ok {
+									if tlsData, ok := docker["TLSData"]; ok && tlsData != nil {
+										usesTLS = true
+									}
+								}
+							}
+						}
+					}
+				}
+
+				contexts = append(contexts, ContextInfo{
+					Name:        ctx.Name,
+					Current:     ctx.Current,
+					Description: ctx.Description,
+					DockerHost:  ctx.DockerEndpoint,
+					TLS:         usesTLS,
+				})
+				lines = []byte{}
+			}
+		} else {
+			lines = append(lines, b)
+		}
+	}
+	// Handle last line if no trailing newline
+	if len(lines) > 0 {
+		var ctx struct {
+			Name           string `json:"Name"`
+			Current        bool   `json:"Current"`
+			Description    string `json:"Description"`
+			DockerEndpoint string `json:"DockerEndpoint"`
+		}
+		if err := json.Unmarshal(lines, &ctx); err != nil {
+			return nil, fmt.Errorf("failed to parse context JSON: %w", err)
+		}
+		contexts = append(contexts, ContextInfo{
+			Name:        ctx.Name,
+			Current:     ctx.Current,
+			Description: ctx.Description,
+			DockerHost:  ctx.DockerEndpoint,
+		})
+	}
+
+	return contexts, nil
+}
+
+// UseContext switches to the specified Docker context
+func UseContext(contextName string) error {
+	cmd := exec.Command("docker", "context", "use", contextName)
+	// Don't output to stdout/stderr to keep UI clean
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to switch context to %s: %w", contextName, err)
+	}
+	return nil
+}
+
+// ValidateContext checks if a context switch would succeed by attempting to connect
+func ValidateContext(contextName string) error {
```

Comment:
I think this method is unused?

Reply:
I actually noticed that the docker's CLI version allows switching to unusable contexts, and it was a pain.